src/ast/query.rs

```diff
@@ -803,32 +818,48 @@ impl fmt::Display for TableFactor {
                 Ok(())
             }
             TableFactor::Pivot {
-                name,
-                table_alias,
+                table,
                 aggregate_function,
                 value_column,
                 pivot_values,
-                pivot_alias,
+                alias,
             } => {
-                write!(f, "{}", name)?;
-                if table_alias.is_some() {
-                    write!(f, " AS {}", table_alias.as_ref().unwrap())?;
-                }
                 write!(
                     f,
-                    " PIVOT({} FOR {} IN (",
+                    "{} PIVOT({} FOR {} IN (",
+                    table,
                     aggregate_function,
                     Expr::CompoundIdentifier(value_column.to_vec())
                 )?;
-                for value in pivot_values {
+                for (idx, value) in pivot_values.iter().enumerate() {
+                    if idx > 0 {
+                        write!(f, ", ")?;
+                    }
                     write!(f, "{}", value)?;
-                    if !value.eq(pivot_values.last().unwrap()) {
+                }
+                write!(f, "))")?;
```

Comment:
You can `display_comma_separated` to write this more succinctly (the same comment applies below as well): 

```suggestion
                write!(
                    f,
                    "{} PIVOT({} FOR {} IN ({})",
                    table,
                    aggregate_function,
                    Expr::CompoundIdentifier(value_column.to_vec()),
                    display_comma_separated(pivot_values)
                )?;
```

Reply:
Updated both places